Firebase.cpp

```diff
@@ -97,18 +105,22 @@ bool Firebase::available() {
   return _http.getStreamPtr()->available();
 }
 
-Firebase::Event Firebase::read(String& event) {
+Firebase& Firebase::read() {
   auto client = _http.getStreamPtr();
-  Event type;;
-  String typeStr = client->readStringUntil('\n').substring(7);
-  if (typeStr == "put") {
-    type = Firebase::Event::PUT;
-  } else if (typeStr == "patch") {
-    type = Firebase::Event::PATCH;
-  } else {
-    type = Firebase::Event::UNKNOWN;
-  }
-  event = client->readStringUntil('\n').substring(6);
+  _event = client->readStringUntil('\n').substring(7);
+  _data = client->readStringUntil('\n').substring(6);
   client->readStringUntil('\n'); // consume separator
-  return type;
+  Serial.println(_event);
+  Serial.println(_data);
+  return *this;
+}
+
+JsonObject& Firebase::create() {
+  return _json.createObject();
+}
+
+JsonObject& Firebase::json() {
+  _json = StaticJsonBuffer<200>();
```

Comment:
not reinitializing the `_json` object is against "best practices` of ArduinoJson, see https://github.com/bblanchon/ArduinoJson/wiki/Avoiding-pitfalls#4-dont-reuse-the-same-jsonbuffer and https://github.com/bblanchon/ArduinoJson/wiki/FAQ#how-to-reuse-a-jsonbuffer

One way to do would be to push back decoding `json` back to the sketch and only expose a way to get the raw body.